Analytical_Engine/Query/IsOutlineRectangular.cs

```diff
@@ -0,0 +1,103 @@
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Analytical.Elements;
+using BH.oM.Geometry;
+using BH.oM.Reflection.Attributes;
+using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+
+namespace BH.Engine.Analytical
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Determines whether a panel is a rectangle")]
+        [Input("panel", "The Panel to check if it is a rectangle")]
+        [Output("bool", "True for rectangular panels or false for non-rectangular panels")]
+        public static bool IsOutlineRectangular<TEdge, TOpening>(this IPanel<TEdge, TOpening> panel)
+            where TEdge : IEdge
+            where TOpening : IOpening<TEdge>
+        {
+            List<ICurve> curves = panel.ExternalEdges.SelectMany(x => x.Curve.ISubParts()).ToList();
+
+            List<PolyCurve> polycurves = Engine.Geometry.Compute.IJoin(curves);
+
+            //Check there is a single continuous curve defining the Panel
+            if (polycurves.Count != 1)
+                return false;
+
+            PolyCurve polycurve = polycurves.First();
+
+            //Check that all subparts of the curve are linear
+            if (polycurve.SubParts().Any(x => !x.IIsLinear()))
+                return false;
+
+            //Group curves by direction vector to obtain discontinuity points
+            List<Point> points = new List<Point>();
+            var groupedCurves = curves.GroupBy(x => x.IStartDir());
```

Comment:
This fails if two parallel edges has the same direction. The Join command should be flipping the edges accordingly, so would use the subcurves of that instead of the original curves.

```suggestion
            var groupedCurves = polycurve.SubParts().GroupBy(x => x.IStartDir());
```